addons/mail/static/src/messaging/entity/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -362,9 +363,23 @@ function MessagingNotificationHandlerFactory({ Entity }) {
 
         /**
          * @private
-         * @param {Object[]} elements
+         * @param {Object} data
          */
-        _handleNotificationPartnerMailFailure(elements) {}
+        _handleNotificationPartnerMessageNotificationUpdate(data) {
+            for (const messageData of data) {
+                const message = this.env.entities.Message.insert(
+                    this.env.entities.Message.convertData(messageData)
+                );
+                // implicit: failures are sent by the server as notification
+                // only if the current partner is author of the message
+                if (!message.author && this.messaging.currentPartner) {
+                    message.update({ author: [['link', this.messaging.currentPartner]] });
+                }
+            }
+            this.env.messaging.notificationGroupManager.computeGroups();
+            // manually force recompute of counter (after computing the groups)
+            this.env.messaging.messagingMenu.update();
```

Comment:
`this.env.messaging` => `this.messaging`

Reply:
ok